src/roiextractors/memmapextractors.py

```diff
@@ -2,109 +2,118 @@
 import shutil
 import tempfile
 from pathlib import Path
+from typing import Union, Optional
 
 import numpy as np
 from tqdm import tqdm
 
 from .extraction_tools import PathType, check_get_frames_args
 from .imagingextractor import ImagingExtractor
 
+PathType = Union[str, Path]
+DtypeType = Union[str, np.dtype]
+OptionalDtypeType = Optional[DtypeType]
+DtypeType = Union[str, np.dtype]
+
 
 class MemmapImagingExtractor(ImagingExtractor):
+
+    extractor_name = "MemmapImagingExtractor"
+
     def __init__(
         self,
-        imaging_extractor: ImagingExtractor,
-        save_path: PathType = None,
-        verbose: bool = False,
+        file_path: PathType,
+        frame_shape: tuple,
+        dtype: DtypeType,
+        offset: int = 0,
+        sampling_frequency: float = 0,
```

Comment:
I'd make sampling frequency required, since there is no global proprietary header for this format

Reply:
At this level I think we could support the ability to extract images even if the ability to map this to times is not available.

Reply:
Dangerous territory; frame rate had to be present to allow acquisition, and core methods such as `frame_to_time` and vice versa won't be possible without it. I thought SpikeInterface and the other extractors here treat `sampling_frequency` generally as a requirement (unless, of course, it is automatically set from a header)?

Reply:
All of those methods themselves will fail for lack of a sampling frequency. What is the danger that you see?

Reply:
> All of those methods themselves will fail for lack of a sampling frequency. What is the danger that you see?

Why accept a value, or default a value for that matter, that will fail further down the line? Also an NWB requirement.


Elsewhere in the ecosystem...

https://github.com/SpikeInterface/spikeinterface/blob/master/spikeinterface/core/baserecording.py#L25

Required.

https://github.com/SpikeInterface/spikeinterface/blob/master/spikeinterface/core/basesorting.py#L13

Required.


Implication being that
https://github.com/catalystneuro/roiextractors/blob/master/src/roiextractors/imagingextractor.py#L24 should also be required.

Reply:
The use case that i mentioned is that you want to extract the images even if you don't know the times associated to each of the frames.  As long as they are using it to check the images, perform operations on them and do anything that does not require associating frames with times, they will be fine. Most of the imaging processing that I know does not require knowing the times.

But in our ecosystem you know the use cases better than me. For `nwb-conversion-tools` for example we need this for synchronization with behavior and other measures so over there it should be required.

If you think that we should prohibit the image processing use case at this level I am fine with that.

Reply:
> The use case that i mentioned is that you want to extract the images even if you don't know the times associated to each of the frames. As long as they are using it to check the images, perform operations on them and do anything that does not require associating frames with times, they will be fine. Most of the imaging processing that I know does not require knowing the times.

Then for what you want here, the best case would be to have the `read_binary` method be a common helper function in `extractor_tools.py` or something like that. That way, if people want to do that, they can use the function to read the binary blob and do what they want.

But anything at the extractors level of operations (and that goes for SpikeInterface, too), requires a notion of temporal scale in order to visualize and package the data in a way that makes sense for analysis or further processing like segmentation.

Reply:
That's fine.